packages/react-instantsearch-core-next/src/__tests__/InstantSearch.test.tsx

```diff
@@ -0,0 +1,129 @@
+import { render } from '@testing-library/react';
+import React, { version as ReactVersion } from 'react';
+
+import { createSearchClient } from '../../../../test/mock';
+import { IndexContext } from '../IndexContext';
+import { InstantSearch } from '../InstantSearch';
+import { InstantSearchContext } from '../InstantSearchContext';
+import version from '../version';
+
+import type { InstantSearch as InstantSearchType } from 'instantsearch.js';
+import type { IndexWidget } from 'instantsearch.js/es/widgets/index/index';
+
+describe('InstantSearch', () => {
+  test('renders children', () => {
+    const searchClient = createSearchClient();
+
+    const { container } = render(
+      <InstantSearch indexName="indexName" searchClient={searchClient}>
+        Children
+      </InstantSearch>
+    );
+
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        Children
+      </div>
+    `);
+  });
+
+  test('provides the search instance', () => {
+    const searchClient = createSearchClient();
+    let searchContext: InstantSearchType | null = null;
+
+    render(
+      <InstantSearch indexName="indexName" searchClient={searchClient}>
+        <InstantSearchContext.Consumer>
+          {(context) => {
+            searchContext = context;
+            return null;
+          }}
+        </InstantSearchContext.Consumer>
+      </InstantSearch>
+    );
+
+    expect(searchContext).toEqual(
+      expect.objectContaining({
+        start: expect.any(Function),
+        dispose: expect.any(Function),
+        addWidgets: expect.any(Function),
+        removeWidgets: expect.any(Function),
+      })
+    );
+  });
+
+  test('provides the main index', () => {
+    const searchClient = createSearchClient();
+    let indexContext: IndexWidget | null = null;
+
+    render(
+      <InstantSearch indexName="indexName" searchClient={searchClient}>
+        <IndexContext.Consumer>
+          {(context) => {
+            indexContext = context;
+            return null;
+          }}
+        </IndexContext.Consumer>
+      </InstantSearch>
+    );
+
+    expect(indexContext).toEqual(
+      expect.objectContaining({
+        $$type: 'ais.index',
+        addWidgets: expect.any(Function),
+        removeWidgets: expect.any(Function),
+      })
+    );
+  });
+
+  test('attaches users agents', () => {
+    const searchClient = createSearchClient();
+
+    render(<InstantSearch indexName="indexName" searchClient={searchClient} />);
+
+    expect(searchClient.addAlgoliaAgent).toHaveBeenCalledWith(
+      `react (${ReactVersion})`
+    );
+    expect(searchClient.addAlgoliaAgent).toHaveBeenCalledWith(
+      `react-instantsearch-core (${version})`
```

Comment:
I think there's some regexes that require react-instantsearch, worth having the same as v6 for simplicity

Reply:
Ah I just realized in v6 we add `react-instantsearch`, and not `react-instantsearch-core`. I'll also add a specific one for this package, that I'll call `react-instantsearch-hooks` for now.

Reply:
See afcd66a7c974e03e8d7914753f4711dcbdb156e1 and ba62c3516fac363e83942552a3200dc71654018d.